torch/optim/radam.py

```diff
@@ -60,13 +62,11 @@ def __setstate__(self, state):
             group.setdefault("foreach", None)
             group.setdefault("differentiable", False)
             group.setdefault("decoupled_weight_decay", False)
-        state_values = list(self.state.values())
-        step_is_tensor = (len(state_values) != 0) and torch.is_tensor(
-            state_values[0]["step"]
-        )
-        if not step_is_tensor:
-            for s in state_values:
-                s["step"] = torch.tensor(float(s["step"]), dtype=_get_scalar_dtype())
+            group.setdefault("capturable", False)
+            for p in group["params"]:
+                p_state = self.state.get(p, [])
+                if len(p_state) != 0 and not torch.is_tensor(p_state['step']):
+                    p_state["step"] = torch.tensor(float(p_state["step"]), dtype=_get_scalar_dtype())
```

Comment:
step needs to be on the param's device if capturable is on and on CPU otherwise

Reply:
I copied this from Adam?

Reply:
this is the non-tensor case. So I don't think this applies right? (at least I'm guessing)

Reply:
no this applies, it should get promoted to a tensor correctly. curr adam is bad and is getting fixed in my pr

Reply:
I'll update